packages/date/src/index.js

```diff
@@ -357,18 +357,21 @@ export function format( dateFormat, dateValue = new Date() ) {
 }
 
 /**
- * Formats a date (like `date()` in PHP), in the site's timezone.
+ * Formats a date (like `date()` in PHP).
  *
- * @param {string}                           dateFormat PHP-style formatting string.
- *                                                      See php.net/date.
- * @param {(Date|string|Moment|null)}        dateValue  Date object or string,
- *                                                      parsable by moment.js.
+ * @param {string}                      dateFormat PHP-style formatting string.
+ *                                                 See php.net/date.
+ * @param {(Date|string|Moment|null)}   dateValue  Date object or string, parsable
+ *                                                 by moment.js.
+ * @param {(string|number|null)}        timezone   Timezone to output result in or
+ *                                                 a UTC offset.
+ *                                                 Defaults to timezone from site.
+ *                                                 See momentjs.com.
```

Comment:
Reading this, it's not immediately obvious to me how to proceed to find more information from the homepage of Moment.js.

A few thoughts:

- The documentation generator has special treatment for the JSDoc `@see` tag, insofar as creating a special "Related resources" section ([example](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/README.md#BlockPreview)). It may be simpler to include a longer/full URL here.
- I ask this in part because, speaking personally, I'm still not sure what format we're expecting here. Is it the identifier used in arguments of [moment.tz](https://momentjs.com/timezone/docs/#/using-timezones/parsing-in-zone/)?
   1. `moment.tz` is technically a plugin to (not core of) the Moment.js library, so directing to [the documentation of Moment.js](https://momentjs.com/docs/) would not be sufficient.
   2. As noted in the above `moment.tz` link, these formats aren't necessarily specific to Moment.js, but rather a standardized notation ([referenced Wikipedia article](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones)). We should prefer a link to the source.
   3. Or at least, I'm motivated to avoid links to Moment.js because I'd personally jump at the first opportunity to shed the 50kb+ of (largely unused) Moment code.

Reply:
I agree. Even though we internally use the library moment, it'd be better if we could use some "generic references" on valid timezones and offsets. Following your advice, I removed the explicit mention to momentjs’ homepage and linked to Wikipedia articles in c3d5fe4:

* [Timezones](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones)
* [UTC Offsets](https://en.wikipedia.org/wiki/ISO_8601#Time_offsets_from_UTC)

However, it's important to keep in mind that the package already exposes the fact that it's internally compatible with momentjs, as `dateValue` params can be a `Moment`.

Reply:
> However, it's important to keep in mind that the package already exposes the fact that it's internally compatible with momentjs, as `dateValue` params can be a `Moment`.

Right, that's fine. I don't think we need to mislead by pretending that we don't support this. In retrospect, we perhaps should have considered to avoid this support in the first place, but the time for that discussion has come and gone. At this point, for me, it's more about: If we _can_ avoid references by including something more generic/canonical, then we should.